src/client/components/app/Transactions/SetRatesTx.tsx

```diff
@@ -0,0 +1,264 @@
+  header: string;
+  onClose: () => void;
+  onSelectedCreditLineChange: Function;
+  onPositionChange: (data: { credit?: string; amount?: string }) => void;
+}
+
+interface SetRateProps {
+  header: string;
+  onClose: () => void;
+  acceptingOffer?: boolean;
+  onSelectedCreditLineChange: Function;
+  onPositionChange: (data: { credit?: string; amount?: string }) => void;
+}
+
+export const SetRatesTx: FC<SetRateProps> = (props) => {
+  const userMetadata = useAppSelector(LinesSelectors.selectUserPositionMetadata);
+  const { t } = useAppTranslation('common');
+  const dispatch = useAppDispatch();
+  const [transactionCompleted, setTransactionCompleted] = useState(0);
+  const { header, onClose, onPositionChange } = props;
+  const [transactionLoading, setLoading] = useState(false);
+  const [targetAmount, setTargetAmount] = useState('1');
+  const [errors, setErrors] = useState<string[]>(['']);
+  const selectedCredit = useAppSelector(LinesSelectors.selectSelectedLine);
+  const selectedPosition = useAppSelector(LinesSelectors.selectSelectedPosition);
+  const walletNetwork = useAppSelector(WalletSelectors.selectWalletNetwork);
+  const setSelectedCredit = (lineAddress: string) => dispatch(LinesActions.setSelectedLineAddress({ lineAddress }));
+  const positions = useAppSelector(LinesSelectors.selectPositionsForSelectedLine);
+  const [drate, setDrate] = useState('0');
+  const [frate, setFrate] = useState('0');
+
+  const acceptingOffer = props.acceptingOffer || (userMetadata.role === BORROWER_POSITION_ROLE && !!selectedPosition);
+
+  if (!selectedPosition) {
+    return null;
+  }
+
+  //Calculate maximum withdraw amount, then humanize for readability
+  const getMaxWithdraw = () => {
+    if (!selectedPosition) {
+      setErrors([...errors, 'no selected position']);
+      return '0';
+    }
+    const deposit: BigNumber = BigNumber.from(selectedPosition.deposit);
+    const principal: BigNumber = BigNumber.from(selectedPosition.principal);
+    const interestRepaid: BigNumber = BigNumber.from(selectedPosition.interestRepaid);
+    const maxWithdrawAmount = deposit.sub(principal).add(interestRepaid);
+    const maxWithdrawAmountLessDust = maxWithdrawAmount.gte(1) ? maxWithdrawAmount.sub(1) : BigNumber.from(0);
+    const maxWithdrawLessDustNormalized = normalize(
+      'amount',
+      maxWithdrawAmountLessDust.toString(),
+      selectedPosition.token.decimals
+    );
+    return maxWithdrawLessDustNormalized;
+  };
+
+  //Used to determine if amount user wants to withdraw is valid
+  const isWithdrawable = () => {
+    if (!selectedPosition) {
+      return;
+    }
+    const maxWithdrawAmount = Number(toWei(getMaxWithdraw()!, selectedPosition.token.decimals));
+    const amountToWithdraw = Number(toWei(targetAmount, selectedPosition.token.decimals));
+
+    return amountToWithdraw > maxWithdrawAmount;
+  };
+
+  const _updatePosition = () =>
+    onPositionChange({
+      credit: selectedCredit?.id,
+      amount: targetAmount,
```

Comment:
why r u updating amount in SetRates?

Reply:
again, lots of copy pasta. will address in later PR